src/store/consts/utils.ts

```diff
@@ -506,6 +520,49 @@ export const nearestSpacingMultiplicity = (centerTick: number, spacing: number)
   )
 }
 
+export const calculateSqrtPriceFromBalance = (
+  balance: number,
+  spacing: bigint,
+  isXtoY: boolean,
+  xDecimal: bigint,
+  yDecimal: bigint
+) => {
+  const minTick = getMinTick(spacing)
+  const maxTick = getMaxTick(spacing)
+
+  const basePrice = Math.max(
```

Comment:
What's this instruction actually supposed to do? The asymmetry here seems suspicious (only `Math.max`)

Reply:
I have based on `calculateTickFromBalance` function which has similar beginning to get `primaryUnitsPrice`.  `basePrice` is validated price that is inside the range of min and max ticks.

![image](https://github.com/invariant-labs/webapp-a0/assets/93533892/cb90cd1c-6788-4e02-a6c4-d0db47eec4c5)